backend/x11/backend.c

```diff
@@ -606,22 +651,26 @@ struct wlr_backend *wlr_x11_backend_create(struct wl_display *display,
 		return false;
 	}
 
-	if (!query_dri3_formats(x11)) {
-		wlr_log(WLR_ERROR, "Failed to query supported DRI3 formats");
+	if (!query_formats(x11)) {
+		wlr_log(WLR_ERROR, "Failed to query supported DRM formats");
 		return false;
 	}
 
-	const struct wlr_drm_format *dri3_format =
-		wlr_drm_format_set_get(&x11->dri3_formats, x11->x11_format->drm);
-	if (dri3_format == NULL) {
+	const struct wlr_drm_format *x11_format =
+		wlr_drm_format_set_get(&x11->shm_formats, x11->x11_format->drm);
```

Comment:
This effectively disables DRI3 format modifiers when SHM is available. `shm_formats` will never have modifiers.

Maybe we can move the logic to pick a `x11_format` up to the `if (x11->have_dri3)` block? `x11_format` is the format used for allocations, so it should come from DRI3 if we're using a GBM allocator, or from SHM if we're using the shm allocator.

In other words, we shouldn't use `shm_formats` when using DRI3, or vice-versa.

Reply:
I changed it to use a single format set for formats that can be used with either SHM or DRI3, not sure if `importable_formats` is a good name for that.

Reply:
Hm, I'd prefer to have the two sets separate. In the future the allocator won't be initialized in the backend impl anymore, and I'd like to be able to switch the X11 window depth at runtime.

Reply:
I think it's better now, still unsure about the name, having two `x11_format`s doesn't seem great so I went with `pixmap_format`.